stablehlo/reference/Ops.cpp

```diff
@@ -68,6 +81,49 @@ Tensor eval(FloorOp op, const Tensor &operand) {
   return result;
 }
 
+Tensor eval(IotaOp op) {
+  Tensor result(op.getType());
+  Type elType = result.getType().getElementType();
+  if (!isSupportedIntegerType(elType) && !isSupportedFloatType(elType) &&
+      !isSupportedComplexType(elType))
+    report_fatal_error(invalidArgument("Unsupported element type: %s",
+                                       debugString(elType).c_str()));
+  uint64_t iotaDimension = op.getIotaDimension();
+  llvm::RoundingMode rm = APFloat::rmNearestTiesToEven;
+  bool roundingErr;
+  for (auto it = result.index_begin(); it != result.index_end(); ++it) {
+    auto iota = (*it)[iotaDimension];
+    if (isSupportedIntegerType(elType)) {
+      if (isSupportedSignedIntegerType(elType)) {
+        result.set(*it, Element(elType, APInt(elType.getIntOrFloatBitWidth(),
+                                              iota, true)));
+      } else if (isSupportedUnsignedIntegerType(elType)) {
+        result.set(*it, Element(elType, APInt(elType.getIntOrFloatBitWidth(),
+                                              iota, false)));
+      }
+    } else if (isSupportedFloatType(elType)) {
+      APFloat val = APFloat((double)iota);
+      if (elType.isBF16()) val.convert(APFloat::BFloat(), rm, &roundingErr);
+      if (elType.isF16()) val.convert(APFloat::IEEEhalf(), rm, &roundingErr);
+      if (elType.isF32()) val.convert(APFloat::IEEEsingle(), rm, &roundingErr);
+      if (elType.isF64()) val.convert(APFloat::IEEEdouble(), rm, &roundingErr);
```

Comment:
`val.convert(elType.getFloatSemantics(), rm, &roundingErr)`

Reply:
Thanks! I was looking for ways to simplify this, and `FloatType` had the function you suggested `getFloatSemantics()`.